### PR TITLE
Scripts/Instance: The Vortex Pinnacle

### PR DESCRIPTION
**Pull request by Mihc3**
Gonna continue this after HoO.

**Major:**
- [ ] [Slipstream](http://www.wowhead.com/npc=45455): Passenger won't exit current Slipstream vehicle when jumping to another one, Slipstream will enter Slipstream instead.
- [ ] [Slipstream Landing Zone](http://www.wowhead.com/npc=45504): Fix after Slipstream.
Note how it should work: https://www.youtube.com/watch?v=znXLVzbsdnw

**Minor:**
- [ ] [Howling Gale](http://www.wowhead.com/spell=85084): Spell won't proc or it's not scripted well.
- [ ] [Grounding Field](http://www.wowhead.com/npc=47085): Server won't assign script to it, but NullCreatureAI. Why?

### CODE REVIEW

**Comment by jaredjones** [comment]
If I'm not mistaken TC is only accepted patches onto the 6.x and 3.3.5a branches.

**Comment by DDuarte** [comment]
We take PRs to the 4.3.4 branch as well but they will 'need' to be ported (by us) to the other ones (where applicable).

**Comment by jaredjones** [comment]
Ah, gotcha :+1:

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_altairus.cpp [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+class npc_air_current : public CreatureScript
+{
+public:
+    npc_air_current() : CreatureScript("npc_air_current") { }
+
+    struct npc_air_currentAI : public ScriptedAI
+    {
+        npc_air_currentAI(Creature* creature) : ScriptedAI(creature)
+        {
+            me->SetDisableGravity(true);
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_altairus.cpp]: to this by changing inhabittype

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_altairus.cpp [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        }
+
+        void Register() override
+        {
+            BeforeCast += SpellCastFn(spell_upwind_of_altairus_SpellScript::GetDistance);
+            OnEffectHitTarget += SpellEffectFn(spell_upwind_of_altairus_SpellScript::CheckDistance, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
+        }
+
+    private:
+        float upwindDistance;
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_altairus.cpp]: uninitialized by default

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+                stormTargetA->CastStop();
+                stormTargetB->CastStop();
+                stormTargetC->CastStop();
+            }
+
+        private:
+            GuidVector stormTargetGUIDs;
+            Creature* stormTargetA;
+            Creature* stormTargetB;
+            Creature* stormTargetC;
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp]: NO

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+                return;
+            
+            events.ScheduleEvent(EVENT_SOTS_SUMMON, 400);
+        }
+
+        void JustSummoned(Creature* creature) override
+        {
+            if (InstanceScript* instance = me->GetInstanceScript())
+                if (Creature* asaad = instance->GetCreature(DATA_ASAAD))
+                    ENSURE_AI(AsaadAI, asaad->AI())->JustSummoned(creature);
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp]: useless ai cast

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp [hunk 5]
```diff
@@ -0,0 +1,775 @@
...
+                        me->DespawnOrUnsummon(700);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        AsaadAI* asaadAI;
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp]: again no

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp [hunk 6]
```diff
@@ -0,0 +1,775 @@
...
+            Creature* GetSelectedStormTarget(uint32 point)
+            {
+                if (point == POINT_STORM_A)
+                    return stormTargetA;
+                if (point == POINT_STORM_B)
+                    return stormTargetB;
+                if (point == POINT_STORM_C)
+                    return stormTargetC;
+                return NULL;
+            }
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp]: deal with uint64 UnitAI::GetGUID(int32 /*id*/ = 0) const { return 0; } to get rid of these ai casts in other scripts

File: src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp [hunk 7]
```diff
@@ -0,0 +1,775 @@
...
+public:
+    spell_sots_targeting() : SpellScriptLoader("spell_sots_targeting") { }
+
+    class spell_sots_targeting_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_sots_targeting_SpellScript);
+
+        void SelectRandom(std::list<WorldObject*>& targets)
+        {
+            Trinity::Containers::RandomResizeList(targets, 1);
```
Comment (joschiwald) [inline src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp]: `Trinity::Containers::SelectRandomContainerElement` is more efficient

**Comment by Carbenium** [comment]
@Mihapro: anything new on this?

**Comment by Mihapro** [comment]
Started rewriting Asaad

**Comment by Mihapro** [comment]
I have a problem getting closest Storm Target with/without specific aura. Any ideas?
Trinity::NearestCreatureEntryWithLiveStateInObjectRangeCheck + Trunity::UnitAuraCheck
I shouldn't cast any spell on it, just move npc_unstable_grounding_field to it, so conditions won't help. Sniffs also don't tell how second and third Storm Targets are decided. Should I use server-side spells or somethig? To get second closest without beam aura?

**Comment by DDuarte** [comment]
TODO: Port to 6.x. (PR was closed automatically by a branch deletion.)

**Comment by Aokromes** [comment]
more than 1 year and still rooting. anyone can open a new pr for 6.x branch?

**Comment by Aokromes** [comment]
git apply k:\14229.diff
k:\14229.diff:87: trailing whitespace.
(13, 1, 86926, 0, 0, 31, 0, 3, 43875, 0, 0, 0, 0, '', 'SOTS Trigger targets BOSS_ASAAD'),
k:\14229.diff:234: trailing whitespace.
(4599101, 8, -807.351,  498.323,  657.6155, 0, 0, 0, 0, 100, 0),
k:\14229.diff:403: trailing whitespace.
    SPELL_CALL_THE_WIND_CHANNEL = 88772, // visual channeling targeting Air Current
k:\14229.diff:683: trailing whitespace.
            if (GetCaster()->GetExactDist(target) < upwindDistance)
k:\14229.diff:975: trailing whitespace.

error: patch failed: src/server/game/Scripting/ScriptLoader.cpp:350
error: src/server/game/Scripting/ScriptLoader.cpp: patch does not apply
error: patch failed: src/server/game/Spells/SpellMgr.cpp:1
error: src/server/game/Spells/SpellMgr.cpp: patch does not apply
error: src/server/scripts/Kalimdor/CMakeLists.txt: No such file or directory

**Pull request closed without merging**